### PR TITLE
chore: remove deprecated fiels in NSTemplateTier

### PR DESCRIPTION
**Pull request by xcoulon**
## Description
remove deprecated `Template` field in `NSTemplateTierNamespace`
and `NSTemplateTierClusterResources`

## Checks
1. Have you run `make generate` target? **yes**

2. Does `make generate` change anything in other projects (host-operator, member-operator, toolchain-common)? **yes** 

3. In case other projects are changed, please provides PR links.
    - host-operator: https://github.com/codeready-toolchain/host-operator/pull/204
    - toolchain-common: https://github.com/codeready-toolchain/toolchain-common/pull/105

Fixes https://issues.redhat.com/browse/CRT-502

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

A few sentences describing the overall goals of the pull request's commits.

### CODE REVIEW

**Comment by MatousJobanek** [comment]
@xcoulon could you please remove also the `Revision` field as well as the `Type` field?

File: pkg/apis/toolchain/v1alpha1/nstemplatetier_types.go [hunk 1]
```diff
@@ -25,31 +24,14 @@ type NSTemplateTierSpec struct {
...
-
-	// The revision of the corresponding template
-	Revision string `json:"revision"`
-
-	// Template contains an OpenShift Template to be used for namespace provisioning
-	Template templatev1.Template `json:"template"`
-
 	// TemplateRef The name of the TierTemplate resource which exists in the host cluster and which contains the template to use
-	// +optional
 	TemplateRef string `json:"templateRef,omitempty"`
```
Comment (alexeykazakov) [inline pkg/apis/toolchain/v1alpha1/nstemplatetier_types.go:28]:
```suggestion
	TemplateRef string `json:"templateRef"`
```

File: pkg/apis/toolchain/v1alpha1/nstemplatetier_types.go [hunk 2]
```diff
@@ -25,31 +24,14 @@ type NSTemplateTierSpec struct {
...
 
-	// The revision of the corresponding template
-	Revision string `json:"revision"`
-
-	// Template contains an OpenShift Template to be used for provisioning of cluster-scoped resources
-	Template templatev1.Template `json:"template"`
-
 	// TemplateRef The name of the TierTemplate resource which exists in the host cluster and which contains the template to use
-	// +optional
 	TemplateRef string `json:"templateRef,omitempty"`
```
Comment (alexeykazakov) [inline pkg/apis/toolchain/v1alpha1/nstemplatetier_types.go:35]:
```suggestion
	TemplateRef string `json:"templateRef"`
```

**Comment by xcoulon** [comment]
thanks for the suggestions @alexeykazakov!

**Pull request merged**